archive/staticsource.go

```diff
@@ -29,8 +36,20 @@ func (s *staticSource) OrderInfo() (field.Static, bool) {
 }
 
 func (s *staticSource) SendSources(ctx context.Context, zctx *resolver.Context, sf driver.SourceFilter, srcChan chan driver.SourceOpener) error {
+	si := SpanInfo{Span: s.sis.Span}
+	for _, p := range s.sis.ChunkRelativePaths {
+		tsd, _, id, ok := parseChunkRelativePath(p)
+		if !ok {
+			return zqe.E(zqe.Invalid, "invalid chunk path: %v", p)
+		}
+		md, err := readChunkMetadata(ctx, chunkMetadataPath(s.ark, tsd, id))
+		if err != nil {
+			return err
+		}
+		si.Chunks = append(si.Chunks, md.Chunk(id))
+	}
 	so := func() (driver.ScannerCloser, error) {
-		return newSpanScanner(ctx, s.ark, zctx, sf.Filter, sf.FilterExpr, s.si)
+		return newSpanScanner(ctx, s.ark, zctx, sf.Filter, sf.FilterExpr, si)
```

Comment:
Another 1502 conflict here, also with driver.Source
Are you thinking we should replace driver.Source with SpanInfoSource?